addons/survey/models/survey_survey.py

```diff
@@ -517,6 +517,23 @@ def _get_next_page_or_question(self, user_input, page_or_question_id, go_back=Fa
 
         return Question
 
+    def _is_last_page_or_question(self, user_input, page_or_question):
+        pages_or_questions = self._get_pages_or_questions(user_input)
+        current_page_index = pages_or_questions.ids.index(page_or_question.id)
+        next_page_or_question_candidates = pages_or_questions[current_page_index + 1:]
+        if next_page_or_question_candidates:
```

Comment:
I wonder if we should clearly separate cases where conditional questions are configured or not?
I'm guessing your method works either way so maybe not necessary.

Reply:
Not necessary, this will work in any case. Unless we want to speed up computation. -> If no conditional questions and question_index == len(question_ids) - 1 : bam !
Meh'